src/main/client_context.cpp

```diff
@@ -73,8 +74,39 @@ struct ActiveQueryContext {
 	BaseQueryResult *open_result = nullptr;
 };
 
+#ifdef DEBUG
+struct DebugClientContextState : public ClientContextState {
+	~DebugClientContextState() override {
+		D_ASSERT(active_transaction == 0);
+		D_ASSERT(active_query == 0);
+	}
+
+	idx_t active_transaction = 0;
+	idx_t active_query = 0;
+
+	void QueryBegin(ClientContext &context) override {
+		active_query++;
+	}
+	void QueryEnd(ClientContext &context) override {
+		active_query--;
```

Comment:
Can we assert here that `active_query == 1`. This is to ensure that begin of a new query event only happens after the previous one was marked as completed.

If we allow event histories such as 
```
QueryBegin
QueryBegin
QueryEnd
QueryEnd
```
we need a way to distinguish between the different QueryBegin / QueryEnd invocations (i.e. some form of identity) so that we can match them up with the correspnding End event.

Reply:
Makes sense, I've turned them into booleans now and check on every call that they are set up correctly